components/admin_console/system_roles/system_role/system_role_permissions.test.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import {TestHelper} from 'utils/test_helper';
+
+import {readAccess, writeAccess} from './types';
+
+import SystemRolePermissions from './system_role_permissions';
+import SystemRolePermission from './system_role_permission';
+
+describe('admin_console/system_role_permissions', () => {
+    const props = {
+        isLicensedForCloud: false,
+        updatePermissions: jest.fn(),
+        permissionsToUpdate: {
+            environment: readAccess,
+            plugins: writeAccess,
+            site: writeAccess,
+        },
+        role: TestHelper.getRoleMock(),
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(
+            <SystemRolePermissions
+                {...props}
+            />);
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should match snapshot with isLicensedForCloud = true', () => {
+        const wrapper = shallow(
+            <SystemRolePermissions
+                {...props}
+                isLicensedForCloud={true}
+            />);
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('ensure that when you change a prop and component is re-rendered, SystemRolePermission is not being deleted due to isLicensedForCloud being false (test for bug MM-31403)', () => {
```

Comment:
@sbishel This is unit test for https://mattermost.atlassian.net/browse/MM-31403 which we discussed about writing